inspirehep/modules/migrator/tasks.py

```diff
@@ -187,11 +187,6 @@ def add_citation_counts():
             es,
             query={
                 "_source": "references.recid",
-                "filter": {
-                    "exists": {
-                        "field": "references.recid"
-                    }
-                },
```

Comment:
I don't think this is necessary to solve the bug. Actually I realize the best way would really be to exploit the `citations_lookup`.

You should rather create the `records_to_update` list via a generator independent from the input.:
```python
def get_records_to_update_generator(citation_lookup):
    for recid, citation_count in citation_lookup.iteritems():
        yield {'_op_type': 'update',
                     '_index': 'hep',
                     '_type': 'record',
                     '_id': recid,
                     'doc': {'citation_count': citation_count}
                 }

for chunk in chunker(get_records_to_update_generator(citations_lookup), CHUNK_SIZE):
    es_bulk(es, chunk)
```

With the current implementation you were risking of filling up the memory with a `records_to_update` list with 1.5M records :smile: .